tests/libs/util/socket_helper.cpp

```diff
@@ -32,25 +39,49 @@ get_address_from_string(
     freeaddrinfo(address_info);
 }
 
-_base_socket::_base_socket(int _sock_type, int _protocol, uint16_t _port)
-    : socket(INVALID_SOCKET), sock_type(_sock_type), protocol(_protocol), port(_port), local_address{},
-      local_address_size(sizeof(local_address))
+std::string
+get_string_from_address(_In_ const void* sockaddr, ADDRESS_FAMILY family)
+{
+    char ip_string[MAXIMUM_IP_BUFFER_SIZE] = {0};
+    if (family == AF_INET) {
+        struct sockaddr_in* addr = (struct sockaddr_in*)sockaddr;
+        InetNtopA(AF_INET, &addr->sin_addr, ip_string, MAXIMUM_IP_BUFFER_SIZE);
+        return std::string(ip_string);
+    } else {
+        struct sockaddr_in6* addr = (struct sockaddr_in6*)sockaddr;
+        InetNtopA(AF_INET6, &addr->sin6_addr, ip_string, MAXIMUM_IP_BUFFER_SIZE);
+        return std::string(ip_string);
+    }
```

Comment:
Could this be simplified to:
```suggestion
    DWORD length = sizeof(ip_string);
    WSAAddressToStringA(sockaddr, INET_ADDR_LENGTH(((PSOCKADDR)sockaddr)->sa_family), nullptr, ip_string, &length);
    return std::string(ip_string);
```
?

Reply:
Fixed.